app/graphql/connections/active_record_cursor_paginate_connection.rb

```diff
@@ -0,0 +1,134 @@
+        raise GraphQL::ExecutionError, 'When using `before` you must also provide `last` if also providing `first`.'
+      end
+    end
+
+    # Retrieve the nodes for the current page.
+    # @return [Array<Object>] The list of nodes for the current page.
+    def nodes # rubocop:disable Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+      load_nodes
+      records = if @inverted
+                  @nodes.records.reverse
+                else
+                  @nodes.records
+                end
+
+      if before.nil? && first.present? && last.present? && last < first
+        records.last(last)
+      elsif after.nil? && last.present? && first.present? && first < last
+        records.first(first)
+      else
+        records
+      end
+    end
+
+    # Check if there is a previous page of results.
+    # @return [Boolean] True if there is a previous page, false otherwise.
+    def has_previous_page # rubocop:disable Naming/PredicatePrefix,Naming/PredicateMethod
+      load_nodes
+      @nodes.has_previous?
+    end
+
+    # Check if there is a next page of results.
+    # @return [Boolean] True if there is a next page, false otherwise.
+    def has_next_page # rubocop:disable Naming/PredicatePrefix,Naming/PredicateMethod
+      load_nodes
+      @nodes.has_next?
+    end
+
+    # Get the cursor for a specific item.
+    # @param item [Object] The item to get the cursor for.
+    # @return [String] The cursor for the item.
+    def cursor_for(item)
+      load_nodes
+      @nodes.cursor_for(item)
+    end
+
+    private
+
+    # Load the nodes using cursor-based pagination.
+    # @return [ActiveRecordCursorPaginate::Result] The paginated result set.
+    def load_nodes
+      @nodes ||= begin # rubocop:disable Naming/MemoizedInstanceVariableName
+        paginator = items.cursor_paginate
+
+        set_order(paginator)
+        set_limit(paginator)
+
+        if after.present?
+          paginator.after = after
+        elsif before.present?
+          paginator.before = before
+        end
+
+        paginator.fetch
+      end
+    end
+
+    # Set the order for the paginator based on the provided order arguments.
+    # @param paginator [ActiveRecordCursorPaginate::Paginator] The paginator to set the order for.
+    # @return [void]
+    def set_order(paginator) # rubocop:disable Metrics/AbcSize,Naming/AccessorMethodName
+      order_arguments = if arguments.key?(:order_by) && arguments[:order_by].present?
+                          if arguments[:order_by].direction.present?
+                            { arguments[:order_by].field => resolve_direction(arguments[:order_by].direction.to_sym) }
+                          else
+                            { arguments[:order_by].field => resolve_direction(:asc) }
+                          end
+                        else
+                          { created_at: resolve_direction(:asc) }
+                        end
+
+      paginator.order = order_arguments
```

Comment:
Assign the arguments directly to `paginator.order`  to keep consistency with how `set_limit` reads.

Reply:
Fixed in be89e22